src/runner.ts

```diff
@@ -408,6 +412,22 @@ export class ResolveRunner implements Types.IResolveRunner {
       if (uriResolver) {
         lookupResult.resolved = await uriResolver.resolve(Utils.uriToJSONPointer(ref));
 
+        // support custom transformers
+        if (this.transformDereferenceResult) {
+          const { result, error } = await this.transformDereferenceResult({
+            source: uriResolver.source,
+            result: lookupResult.resolved.result,
+            targetAuthority: ref,
+            parentAuthority: this.baseUri,
+            parentPath,
+            fragment: ref.fragment(),
+          });
+          if (error) {
+            throw new Error(`Could not transform dereferenced result for '${ref.toString()}' - ${String(error)}`);
```

Comment:
Perhaps could set `lookupResult.error` rather than throw?

Reply:
Let me know what you think now after moving where this get's called at.